cli/hclfmt.go

```diff
@@ -19,10 +19,18 @@ import (
 // runHCLFmt recursively looks for terragrunt.hcl files in the directory tree starting at workingDir, and formats them
 // based on the language style guides provided by Hashicorp. This is done using the official hcl2 library.
 func runHCLFmt(terragruntOptions *options.TerragruntOptions) error {
-	terragruntOptions.Logger.Printf("Formatting terragrunt.hcl files from the directory tree %s.", terragruntOptions.WorkingDir)
 
+	var tgHclFiles []string
+	var err error
 	workingDir := terragruntOptions.WorkingDir
-	tgHclFiles, err := zglob.Glob(util.JoinPath(workingDir, "**", "*.hcl"))
+	targetFile := terragruntOptions.HclFile
+	if targetFile != "" {
+		terragruntOptions.Logger.Printf("Formatting terragrunt.hcl file at: %s.", targetFile)
+		tgHclFiles, err = zglob.Glob(util.JoinPath(workingDir, targetFile))
+	} else {
+		terragruntOptions.Logger.Printf("Formatting terragrunt.hcl files from the directory tree %s.", terragruntOptions.WorkingDir)
+		tgHclFiles, err = zglob.Glob(util.JoinPath(workingDir, "**", "*.hcl"))
+	}
```

Comment:
I think this whole routine can be simplified to:

```go
targetFile := terragruntOptions.HclFile
if targetFile != "" {
    terragruntOptions.Logger.Printf("Formatting terragrunt.hcl file at: %s.", targetFile)
    return formatTgHCL(terragruntOptions, targetFile)
}

// what the function was before
```

Reply:
mmm yes—i did have to add a line to format the entire path (ensuring working directory is appended to the specified file path). at the moment it's setup so that it's expecting the file's relative path to the working directory. i was planning on specifying this in the docs or change it to require an absolute path

Reply:
Ah you can use https://golang.org/pkg/path/filepath/#IsAbs to check if it is absolute first.